plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -111,14 +112,19 @@ export class TaskWorker {
     });
   }
 
-  start() {
-    const recoverTasksId = setInterval(async () => {
+  async recoverTasks() {
+    try {
       const enabled = await this.options.taskBroker.recoverTasks?.();
-      if (!enabled) {
-        clearInterval(recoverTasksId);
+      if (enabled) {
+        setTimeout(async () => await this.recoverTasks(), 60000);
```

Comment:
Apologies for going on about this, but it is passing an async function to either `setTimeout` or `setInterval`, which concerns me.  The `setTimeout` function does not wait for this promise to resolve.  It simply schedules the callback to be executed after the specified delay, and then the execution continues independently.  The code within the callback (after the await expressions) may execute at a significantly different time than you might expect, especially if there are long-running promises.

I've had memory leaks with similar things.

This is what I meant by wrapping `setTimeout` in a promise:

```ts
await new Promise(resolve => setTimeout(resolve, 6000));
await this.recoverTasks();
```

You could even have a `wait` function 

```ts
const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
```

If you think this is fine, please don't hesitate to ignore me.

Or maybe someone else can offer a different opinion?

Reply:
@dagda1 Isn't `setTimeout` called once? 
What I'm doing now is calling `setTimeout`, which calls itself again only after `recoverTasks` was performed.

Reply:
Apologies if I'm not getting my message across.

I think calling `setTimeout` or `setInterval` with an async callback can lead to unreliable behaviours. Neither function waits for the promises to resolve, and the code within the callback can execute at a different time than you expect.

Wrapping `seTimeout` in a promise is a way to avoid the async callback.

If you think this is not needed, then please ignore it.